default/AI/ResearchAI.py

```diff
@@ -453,28 +492,45 @@ def generate_research_orders():
     rng = random.Random()
     rng.seed(fo.getEmpire().name + fo.getGalaxySetupData().seed)
 
-    research_reqs = calculate_research_requirements(empire)
+    calculate_research_requirements(empire)
     total_rp = empire.resourceProduction(fo.resourceType.research)
 
     if total_rp <= 0: # No RP available - no research.
         return
 
+    raw_priorities = {}
     priorities = {}
+    on_path_to = {}
     for tech_name in fo.techs():
-        priority = get_priority(rng, tech_name)
+        this_tech = fo.getTech(tech_name)
+        if not this_tech:
+            continue
+        raw_priorities[tech_name] = get_priority(rng, tech_name)
+    for tech_name, priority in raw_priorities.iteritems():
         if not tech_is_complete(tech_name) and priority >= 0:
-            turn_needed = float(research_reqs[tech_name][1]) / total_rp
-            priorities[tech_name] = float(priority) / turn_needed / max(1, (turn_needed - 12) / 4 + 1)**2
+            turn_needed = max(research_reqs[tech_name][2], math.ceil(float(research_reqs[tech_name][1]) / total_rp))
+            adjusted_priority = float(priority) / turn_needed / max(1, (turn_needed - 12) / 4 + 1)**2
```

Comment:
Then again, after these techs are researched, if I started with the slow techs, I will be able to research the other ones much faster (assuming that our RP are growing in the meantime). The slow but cheap techs will still take maximum time.

So long story short, I don't consider this to be an issue resolved by a simple formula but something which would need a more complex approach which at least at this point in time is probably not worth it as this is a second order considerations while we have not solved most of the first-order problems yet...

So for now, I think we should ignore these timing considerations or at least not restrain a tech by long research time unless we find a very specific problem with the current tech tree that is related to this issue.